cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisionerNotifier.java

```diff
@@ -102,15 +103,25 @@ public void deprovisioning(ProgramRunId programRunId) {
   }
 
   public void deprovisioned(ProgramRunId programRunId) {
+    deprovisioned(programRunId, System.currentTimeMillis());
+  }
+
+  public void deprovisioned(ProgramRunId programRunId, long endTimestamp) {
     publish(ImmutableMap.of(
       ProgramOptionConstants.PROGRAM_RUN_ID, GSON.toJson(programRunId),
-      ProgramOptionConstants.CLUSTER_STATUS, ProgramRunClusterStatus.DEPROVISIONED.name()));
+      ProgramOptionConstants.CLUSTER_STATUS, ProgramRunClusterStatus.DEPROVISIONED.name(),
+      ProgramOptionConstants.CLUSTER_END_TIME, String.valueOf(endTimestamp)));
```

Comment:
Use `Long.toString(endTimestamp)` instead of `String.valueOf`. The `valueOf` takes an `Object`, hence there would be an unnecessary boxing.